src/steps/git.rs

```diff
@@ -295,16 +295,24 @@ impl RepoStep {
     async fn pull_repo<P: AsRef<Path>>(&self, ctx: &ExecutionContext<'_>, repo: P) -> Result<()> {
         let before_revision = get_head_revision(&self.git, &repo);
 
-        if ctx.config().verbose() {
-            println!("{} {}", style("Pulling").cyan().bold(), repo.as_ref().display());
-        }
-
         let mut command = AsyncCommand::new(&self.git);
 
-        command
-            .stdin(Stdio::null())
-            .current_dir(&repo)
-            .args(["pull", "--ff-only"]);
+        if ctx.config().git_fetch_only().is_some_and(|f| *f) {
+            if ctx.config().verbose() {
+                println!("{} {}", style("Fetching").cyan().bold(), repo.as_ref().display());
+            }
+
+            command.stdin(Stdio::null()).current_dir(&repo).args(["fetch"]);
+        } else {
+            if ctx.config().verbose() {
+                println!("{} {}", style("Pulling").cyan().bold(), repo.as_ref().display());
+            }
+
+            command
+                .stdin(Stdio::null())
+                .current_dir(&repo)
+                .args(["pull", "--ff-only"]);
+        }
```

Comment:
Looks like the submodule update procedure should also be considered as a part of `git pull`?

Reply:
Let's update `pull_repos()` function as well

1. Rename it to `pull_or_fetch_repos()`
2. Update the doc
3. Update the outputs exclusive to `git pull`